lib/https.js

```diff
@@ -34,6 +34,12 @@ const { urlToOptions, searchParamsSymbol } = require('internal/url');
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
+  }
+  opts = opts ? util._extend({}, opts) : {};
```

Comment:
@refack, And if necessary, we can open a new PR about changing all `util._extend` to `Object.assign` in this file next time. Because it not only contains one `util._extend` here.